content/tutorials/reactor-event-cloudflare.textile

```diff
@@ -0,0 +1,221 @@
+</a>
+
+Fill out the form to provide Ably with the details it needs to invoke your Cloudflare Worker.
+
+Your Reactor event will now be created, and you'll be able to continue developing the game.
+
+h2. Publishing to an Ably Channel when a delivery is made
+
+Open the file "./src/feature/player/movement.js":https://github.com/ably/tutorials/blob/reactor-event-cloudflare-javascript/src/feature/player/movement.js
+
+This file contains the React code for handling player movements, but it also contains a connection to our Ably Channel, firstly, on line 18
+
+```[javascript]
+    window.Ably.connection.on(function(stateChange) {
+        console.log('New connection state is ' + stateChange.current);
+    })
+
+    const outboundChannel = window.Ably.channels.get('cloudflareworkerdemo');
+```
+
+This code establishes a connection to our "cloudflareworkerdemo" channel.
+
+When the player completes a level the @handleWin@ function is called, and on line 64 we publish a message to our channel.
+
+```[javascript]
+  outboundChannel.publish('player', { playerSteps },
+      err => {
+          if (err) {
+              return console.error('Failed to publish', err);
+          }
+          console.log('published');
+      }
+  )
+```
+
+Our message contains the number of moves the player used to deliver the parcel.
+
+h2. Processing the message in the Worker
+
+Ably Reactor is monitoring the channel and calls the Cloudflare Worker with the message contents.
+We can see how this works by taking a look at the Worker file again "./cloudflare-worker/index.js":https://github.com/ably/tutorials/blob/reactor-event-cloudflare-javascript/cloudflare-worker/index.js
+
+```[javascript]
+async function publishToAbly(data) {
+    try {
+        const URL = `https://rest.ably.io/channels/cloudflare-bot/messages?key=${key}`;
+        
+        await fetch(`${URL}`, {
+            method: 'POST',
+            headers: { 'Content-Type': 'application/json' },
+            body: JSON.stringify({ data });
+        });
+
+    } catch (error) {
+        return error;
+    }
+}
+```
+The Worker is re-publishing our initial message to a channel called @cloudflare-bot@ using the "Ably REST API":https://www.ably.io/documentation/rest. "We would usually recommend":https://support.ably.io/a/solutions/articles/3000061848 that you use the "REST Client Library API":https://www.ably.io/documentation/rest rather than make a direct POST to the Ably Rest API, since it is more discoverable and easier to use, however in this particular case, installing the library files in the Worker would mean bundling the entire library with your Worker, which would increase the size of your script unnecessarily. 
+
+We'll subscribe to this channel on our dashboard to receieve these notifications.
+
+h2. Connecting the dashboard
+
+In our game app, there's a React Component called Scoreboard in *./src/feature/scoreboard/index.js* which contains the following code on line 12
```

Comment:
Perhaps instead of saying the line number, it'll be worth saying add it after this statement or method definition?

Reply:
There is nothing to add, the line number is just pointing out where in the code we're looking.